### PR TITLE
Add support of OSMesa (Gallium) headless backend

### LINKED ISSUE #1555: headless tests on travis don't antialias fills
Actual:
image

Expected:
image

### PR DESCRIPTION
**Pull request by brunoabinader**
As per https://github.com/mapbox/mapbox-gl-native/issues/4498#issuecomment-258109635 - using OSMesa as headless backend fixes the rendering artifacts that were causing us to ignore 42 render tests. This matches 1:1 with macOS results, so we can safely unignore them.

Follow-up to work started by @mikemorris in #4638.
Fixes #1555 and #4498.

Caveats:
- We'll need some ❤️ in our `.travis.yml` after this, as e.g. the node build won't need X library dependencies anymore.
- EGL support to be added in a following PR.

Default backends on Linux:
- GLX: `mbgl-glfw` and `mbgl-render`
- OSMesa: `mbgl-node`, `mbgl-test` and `mbgl-benchmark`

### CODE REVIEW

**Comment by brunoabinader** [comment]
Also missing:
- Remove render test ignores - to be properly reviewed on a following PR.

**Comment by brunoabinader** [comment]
Still working on a few things to make sure `mesa` mason package works in both Ubuntu Trusty and Xenial (-cxx11abi).

**Comment by brunoabinader** [comment]
Extracted 29f764b479bc88880b7f097ddccca21e8ac0b0ee from `fix-symbol-clipping` branch to satisfy updated `symbol-features-in/pitched-screen` query test results.

> Fixes #1555 and #4498.

Also fixes  #3175, #1600 and #4593.

**Comment by brunoabinader** [comment]
Tracking gdb hang when attempting to run `make run-test` with OSMesa backend in #6918.

**Comment by brunoabinader** [comment]
This is now ready for review:
- Using OSMesa backend for all Travis builds except Qt 4 and 5.
- Disable gdb check when running coverage tests: see #6918.
- Extracted 29f764b from `fix-symbol-clipping` branch to satisfy updated symbol-features-in/pitched-screen query test results.
- Updated `mapbox-gl-test-suite` to account for the now-passing 42 render tests :tada:

Please review @jfirebaugh @kkaefer @mikemorris.

File: .travis.yml [hunk 1]
```diff
@@ -6,15 +6,13 @@ addons_shortcuts:
   addons_clang35: &clang35
     apt:
       sources:  [ 'ubuntu-toolchain-r-test', 'george-edison55-precise-backports' ]
-      packages: [ 'clang-3.5', 'libstdc++-4.9-dev', 'libstdc++6', 'libllvm3.4', 'cmake', 'cmake-data',
-                  'mesa-utils', 'libxi-dev', 'libglu1-mesa-dev', 'x11proto-randr-dev', 'x11proto-xext-dev', 'libxrandr-dev',
-                  'x11proto-xf86vidmode-dev', 'libxxf86vm-dev', 'libxcursor-dev', 'libxinerama-dev' ]
+      packages: [ 'clang-3.5', 'libstdc++-4.9-dev', 'libstdc++6', 'libllvm3.8v4', 'cmake', 'cmake-data',
+                  'libxrandr-dev', 'libxcursor-dev', 'libxinerama-dev' ]
```
Comment (mikemorris) [inline .travis.yml]: Are these X packages still necessary?

File: platform/default/headless_backend_osmesa.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+void HeadlessBackend::createContext() {
+    if (glContext == nullptr) {
+#if OSMESA_MAJOR_VERSION * 100 + OSMESA_MINOR_VERSION >= 305
+        glContext = OSMesaCreateContextExt(OSMESA_RGBA, 16, 0, 0, nullptr);
+#else
+        glContext = OSMesaCreateContext(OSMESA_RGBA, nullptr);
+#endif
+        if (glContext == nullptr) {
+            Log::Error(Event::OpenGL, "failed to create OSMesa context");
+            OSMesaDestroyContext(glContext);
```
Comment (mikemorris) [inline platform/default/headless_backend_osmesa.cpp]: Is this cleanup necessary if `glContext == nullptr`?

File: scripts/travis_setup.sh [hunk 3]
```diff
@@ -30,27 +30,27 @@ ${CXX} --version
...
+    mapbox_time "start_xvfb" \
+    sh -e /etc/init.d/xvfb start
+    sleep 2 # sometimes, xvfb takes some time to start up
+
+    # Make sure we're connecting to xvfb
+    export DISPLAY=:99.0
+
+    # Install and set up to load a more recent version of mesa
+    mapbox_time "install_mesa" \
+    mason install mesa 11.2.2
```
Comment (mikemorris) [inline scripts/travis_setup.sh]: Should we build a Mesa 13 binary that can be used here as well?

File: .travis.yml (hunk 1 above)
Comment (brunoabinader) [inline .travis.yml]: Yes - these are needed for building the GLFW-based demo application.

File: scripts/travis_setup.sh (hunk 3 above)
Comment (brunoabinader) [inline scripts/travis_setup.sh]: BTW, I'm planning on a posterior PR to rework our Travis workflow e.g. the clang-tidy check shouldn't need gcc nor LLVM libs. Having a GLX-based mesa 13.0.0 could be part of that.

**Pull request merged**